bgpd/bgp_attr_evpn.c

```diff
@@ -217,10 +218,14 @@ void bgp_attr_evpn_na_flag(struct attr *attr, uint8_t *router_flag)
 		if (type == ECOMMUNITY_ENCODE_EVPN &&
 		    sub_type == ECOMMUNITY_EVPN_SUBTYPE_ND) {
 			val = *pnt++;
-			if (val & ECOMMUNITY_EVPN_SUBTYPE_ND_ROUTER_FLAG) {
+
+			if (val & ECOMMUNITY_EVPN_SUBTYPE_ND_ROUTER_FLAG)
 				*router_flag = 1;
-				break;
-			}
+
+			if (val & ECOMMUNITY_EVPN_SUBTYPE_PROXY_FLAG)
+				*proxy = true;
+
+			break;
```

Comment:
Old code iterates through all communities, and break only when it found ECOMMUNITY_EVPN_SUBTYPE_ND_ROUTER_FLAG or all communities are checked. 

New code will break for the first community and exit out of the loop. Is this what you intended to do ?

Reply:
yeah :) I remember thinking about this and wondering why only this parsing was different. If you see other ecom processing, bgp_attr_mac_mobility_seqnum/bgp_attr_rmac/bgp_attr_default_gw, they break out on first occurrence.

It is a minor optimization but really more for consistency.